app/models/search_builder.rb

```diff
@@ -2,8 +2,17 @@
 class SearchBuilder < Blacklight::SearchBuilder
   include Blacklight::Solr::SearchBuilderBehavior
   include BlacklightAdvancedSearch::AdvancedSearchBuilder
-  self.default_processor_chain += [:add_advanced_parse_q_to_solr, :add_advanced_search_to_solr]
+  self.default_processor_chain += %i(add_advanced_parse_q_to_solr
+                                     add_advanced_search_to_solr
+                                     json_api_max_per_page)
   include BlacklightHeatmaps::SolrFacetHeatmapBehavior
 
   include Spotlight::AccessControlsEnforcementSearchBuilder
+
+  JSON_API_MAX = 1_000
+  def json_api_max_per_page(solr_parameters)
+    return unless blacklight_params[:format] == 'json'
+    solr_parameters[:rows] = (blacklight_params[:rows] || blacklight_config.max_per_page).to_i
```

Comment:
Should this be `default_per_page` instead of `max_per_page`?

Maybe, instead of adding another processor, you'd like to override `#rows` instead? Here's the implementation of `rows` upstream (which is used to feed the `rows` parameter in the solr query):

https://github.com/projectblacklight/blacklight/blob/v6.11.2/lib/blacklight/search_builder.rb#L201-L203

Reply:
Ok, I'll give that a try.